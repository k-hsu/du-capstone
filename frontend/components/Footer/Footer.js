import React from "react";
import Flex from "../Flex/Flex";
import Text from "../Text/Text";
import { spacing } from "../../theme";

const Footer = (props) => {
  return (
    <Flex p={`${spacing["0.75"]} ${spacing["4"]}`} {...props}>
      <Text>Hello World</Text>
    </Flex>
  );
};

export default Footer;
