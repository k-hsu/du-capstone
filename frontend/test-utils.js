import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
