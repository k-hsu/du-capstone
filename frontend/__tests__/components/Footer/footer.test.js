import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import Footer from "../../../components/Footer/Footer";

describe("Footer", () => {
  it("should render footer component", () => {
    render(<Footer />);
    expect(screen.getByText("Hello World")).toBeInTheDocument();
  });

  it("should render footer component styling", () => {
    const component = renderer.create(<Footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
