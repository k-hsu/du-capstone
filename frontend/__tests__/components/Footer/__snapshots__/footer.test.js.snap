// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render footer component styling 1`] = `
<div
  className="css-pkc6ac"
  direction="row"
>
  <text
    className="css-yma8y4"
    fontFamily="IBM Plex Sans"
  >
    Hello World
  </text>
</div>
`;
